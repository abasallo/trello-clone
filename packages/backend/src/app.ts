import express from 'express'
import cors from 'cors'
import bodyParser from 'body-parser'

import index from './express/routes/index'
import auth from './express/routes/auth'

import {initialiseSequelize} from './orm/initialiseSequelize'
import {AppModel} from "./orm/model/app.model";

import {initialiseApolloServer} from "./graphql/init";
export const app = express()

export const model: Promise<AppModel> = initialiseSequelize()

initialiseApolloServer(app)
    .then(() => console.log('Apollo Server initialised'))
    .catch((error) => console.error('Apollo Server initialisation error: ' + error))

app.use(cors())

app.use(bodyParser.urlencoded({extended: false}))
app.use(bodyParser.json())

app.use('/', index)
app.use('/auth', auth)

export default app
