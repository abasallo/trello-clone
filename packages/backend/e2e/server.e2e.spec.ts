import * as http from 'http'

import axios from 'axios'

import { ReasonPhrases, StatusCodes } from 'http-status-codes'

import app from '../src/app'

const TEST_SERVER_PORT = 4001

let server: http.Server

beforeAll(async () => {
  server = app.listen({ port: TEST_SERVER_PORT }, () =>
    console.log(`Test Server initialised on port: ${TEST_SERVER_PORT}`)
  )
})

afterAll(() => {
  server.close()
})

let addedBoardId = 4

describe('Server E2E tests', () => {
  describe('Health Check E2E tests', () => {
    test('Expected response for Health Check', async () => {
      const { status, statusText, data } = await axios({
        method: 'get',
        url: `http://localhost:${TEST_SERVER_PORT}/`,
      })

      expect(status).toEqual(StatusCodes.OK)
      expect(statusText).toEqual(ReasonPhrases.OK)

      expect(data).toMatchSnapshot()
    })
  })

  describe('Colour tests', () => {
    test('Expected response for Colour Endpoint', async () => {
      const { status, statusText, data } = await axios({
        method: 'get',
        url: `http://localhost:${TEST_SERVER_PORT}/colour`,
      })

      expect(status).toEqual(StatusCodes.OK)
      expect(statusText).toEqual(ReasonPhrases.OK)

      expect(data.match(/^#[0-9a-f]{3,6}$/i))
    })
  })

  describe('Board E2E tests', () => {
    test('AddBoards', async () => {
      const { status, statusText, data } = await axios({
        method: 'post',
        url: `http://localhost:${TEST_SERVER_PORT}/graphql/`,
        data: {
          query: `
            mutation AddBoard($board: BoardInput!) {
              addBoard(board: $board) {
                id,
                name
              }
            }
          `,
          variables: {
            board: {
              name: 'Board IV',
            },
          },
        },
      })

      expect(status).toEqual(StatusCodes.OK)
      expect(statusText).toEqual(ReasonPhrases.OK)

      addedBoardId = data.data.addBoard.id

      expect(data).toMatchSnapshot({
        data: {
          addBoard: {
            id: expect.any(String),
          },
        },
      })
    })

    test('UpdateBoard', async () => {
      const { status, statusText, data } = await axios({
        method: 'post',
        url: `http://localhost:${TEST_SERVER_PORT}/graphql/`,
        data: {
          query: `
              mutation UpdateBoard($board: BoardInput!) {
                updateBoard(board: $board) {
                  id,
                  name
                }
              }
            `,
          variables: {
            board: {
              id: addedBoardId,
              name: 'Board IV - Updated',
            },
          },
        },
      })

      expect(status).toEqual(StatusCodes.OK)
      expect(statusText).toEqual(ReasonPhrases.OK)

      expect(data).toMatchSnapshot({
        data: {
          updateBoard: {
            id: expect.any(String),
          },
        },
      })
    })

    test('DeleteBoard', async () => {
      const { status, statusText, data } = await axios({
        method: 'post',
        url: `http://localhost:${TEST_SERVER_PORT}/graphql/`,
        data: {
          query: `
          mutation DeleteBoard($id: ID!) {
            deleteBoard(id: $id)
          }
        `,
          variables: {
            id: addedBoardId,
          },
        },
      })

      expect(status).toEqual(StatusCodes.OK)
      expect(statusText).toEqual(ReasonPhrases.OK)

      expect(data).toMatchSnapshot()
    })

    test('GetBoards', async () => {
      const { status, statusText, data } = await axios({
        method: 'post',
        url: `http://localhost:${TEST_SERVER_PORT}/graphql/`,
        data: {
          query: `
              query GetBoards {
                getBoards {
                  id,
                  name
                }
              }
            `,
        },
      })

      expect(status).toEqual(StatusCodes.OK)
      expect(statusText).toEqual(ReasonPhrases.OK)

      expect(data).toMatchSnapshot()
    })
  })
})
