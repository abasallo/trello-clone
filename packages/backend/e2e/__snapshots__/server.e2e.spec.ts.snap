// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Server E2E tests Board E2E tests AddBoards 1`] = `
{
  "data": {
    "addBoard": {
      "id": Any<String>,
      "name": "Board IV",
    },
  },
}
`;

exports[`Server E2E tests Board E2E tests DeleteBoard 1`] = `
{
  "data": {
    "deleteBoard": 1,
  },
}
`;

exports[`Server E2E tests Board E2E tests GetBoards 1`] = `
{
  "data": {
    "getBoards": [
      {
        "id": "1",
        "name": "Board I",
      },
      {
        "id": "2",
        "name": "Board II",
      },
      {
        "id": "3",
        "name": "Board III",
      },
    ],
  },
}
`;

exports[`Server E2E tests Board E2E tests UpdateBoard 1`] = `
{
  "data": {
    "updateBoard": {
      "id": Any<String>,
      "name": "Board IV - Updated",
    },
  },
}
`;

exports[`Server E2E tests Colour tests Expected response for Colour Endpoint 1`] = `"#323b63"`;

exports[`Server E2E tests Health Check E2E tests Expected response for Health Check 1`] = `"OK"`;
